lesson_07/homework_07.py

```diff
@@ -0,0 +1,209 @@
+
+str1 = "The quick brown fox jumps over the lazy dog"
+str2 = "cat"
+print(find_substring(str1, str2))  # поверне -1
+
+# task 7
+# task 8
+# task 9
+# task 10
+"""  Оберіть будь-які 4 таски з попередніх домашніх робіт та
+перетворіть їх у 4 функції, що отримують значення та повертають результат.
+Обоязково документуйте функції та дайте зрозумілі імена змінним.
+"""
+
+# 4 tasks are taken and converted into functions from HW 4:
+
+
+adwentures_of_tom_sawer = """\
+Tom gave up the brush with reluctance in his .... face but alacrity
+in his heart. And while
+the late steamer
+"Big Missouri" worked ....
+and sweated
+in the sun,
+the retired artist sat on a barrel in the .... shade close by, dangled his legs,
+munched his apple, and planned the slaughter of more innocents.
+There was no lack of material;
+boys happened along every little while;
+they came to jeer, but .... remained to whitewash. ....
+By the time Ben was fagged out, Tom had traded the next chance to Billy Fisher for
+a kite, in good repair;
+and when he played
+out, Johnny Miller bought
+in for a dead rat and a string to swing it with—and so on, and so on,
+hour after hour. And when the middle of the afternoon came, from being a
+poor poverty, stricken boy in the .... morning, Tom was literally
+rolling in wealth."""
+
+
+# 07 task = function
+
+
+def count_character_occurrences(text, character):
+    """
+    Counts the number of occurrences of a character in the text "adwentures_of_tom_sawer"
+    :param: text (str): text in which occurrences of the character should be counted
+    character (str): character to be counted
+    :return:
+    int: the number of occurrences of the character in the text
+    """
+    return text.count(character)
+
+
+result = count_character_occurrences(adwentures_of_tom_sawer, 'v')
+print(f'The number of letters "v": {result}')
+
+
+# 08 task = function
+
+
+def count_capital_letters(text):
+    """
+    Counts the number of capital letters in the text "adwentures_of_tom_sawer"
+    :param: text (str): text in which capital letters should be counted
+    :return: int: quantity of capital letters
+    """
+    return sum(1 for char in text if char.isupper())
+
+
+result = count_capital_letters(adwentures_of_tom_sawer)
+print(f'The quantity of capital letters: {result}')
+
+# 09 task = function
+
+import re
+
+
+def split_into_sentences(text):
+    """
+    Splits text into sentences. Uses a regular expression to split the text into sentences,
+    taking into account possible abbreviations and names
+    :param: text: (str): text to split into sentences
+    :return: list: list of sentences
+    """
+    return re.split(r'(?<!\w.\w.)(?<![A-Z][a-z]\.)(?<=\.|\?)\s', adwentures_of_tom_sawer)
+
+
+adwentures_of_tom_sawer_sentences = split_into_sentences(adwentures_of_tom_sawer)
+for sentence in adwentures_of_tom_sawer_sentences:
+    print(f"\n"
+          f"{sentence}")
+
+
+# 10 task = function
+
+
+def starts_with_by_the_time(sentences):

```

Comment:
If you create func that returns bool value - it`s name should start with is_starts_with_by_the_time()